web/src/inbox_ui.ts

```diff
@@ -885,19 +855,77 @@ function show_empty_inbox_channel_view_text(is_empty: boolean): void {
     }
 }
 
+function get_min_load_count(already_rendered_count: number, load_count: number): number {
+    // Since inbox rows have a short height, we need a lot of
+    // rows to fill the screen if the updated row is at the top
+    // of the screen.
+    const extra_rows_for_viewing_pleasure = 50;
+    const ideal_rendered_rows_count = row_focus + extra_rows_for_viewing_pleasure;
+    if (ideal_rendered_rows_count > already_rendered_count + load_count) {
+        return ideal_rendered_rows_count - already_rendered_count;
+    }
+    return load_count;
+}
+
 class InboxTopicListWidget extends TopicListWidget {
     override topic_list_class_name = "inbox-channel-topic-list";
+    topics_widget?: list_widget.ListWidget<topic_list_data.TopicInfo>;
 
-    override build(show_spinner = false): this {
-        const formatter = get_channel_view_formatter(this.get_stream_id());
+    override build(): this {
+        // Hide any existing loading indicators.
+        $("#inbox-loading-indicator .searching-for-more-topics").hide();
         const is_zoomed = true;
-        super.build(show_spinner, formatter, is_zoomed);
+        const $container = $("#inbox-list");
+        const list_info = topic_list_data.get_list_info(
+            this.my_stream_id,
+            is_zoomed,
+            this.filter_topics,
+        );
+
+        const all_topics = list_info.items;
+        this.topics_widget = list_widget.create($container, all_topics, {
+            name: "inbox-channel-topics-list",
+            get_item: list_widget.default_get_item,
+            $parent_container: $("#inbox-view"),
+            modifier_html(item) {
+                const topic_context = format_topic(
+                    item.stream_id,
+                    false,
+                    item.topic_name,
+                    item.unread,
+                    -1,
+                    true,
+                );
+                return render_inbox_row(topic_context);
+            },
+            $simplebar_container: $("html"),
+            is_scroll_position_for_render: views_util.is_scroll_position_for_render,
+            get_min_load_count,
+        });
+
         show_empty_inbox_channel_view_text(this.is_empty());
+        if (!stream_topic_history.is_complete_for_stream_id(this.my_stream_id)) {
+            $("#inbox-loading-indicator .searching-for-more-topics").show();
```

Comment:
Is this the right loading indicator to show?

Reply:
I guess the indicator visible in your screenshot above looks decent.

Reply:
discussion for placement of this is still ongoing but updated to use `z-loader` as that's mutually agreed.

https://chat.zulip.org/#narrow/channel/101-design/topic/user.20research.3A.20stream.20clicks.20in.20Inbox/with/2190388